homeassistant/components/tautulli/sensor.py

```diff
@@ -48,74 +56,78 @@
     }
 )
 
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        icon="mdi:plex",
+        key="tautulli",
+        name="Tautulli",
+        native_unit_of_measurement="Watching",
+    ),
+)
+
 
 async def async_setup_platform(
     hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
+    config: dict,
+    async_add_entities: entity_platform.AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Create the Tautulli sensor."""
-
-    name = config[CONF_NAME]
-    host = config[CONF_HOST]
-    port = config[CONF_PORT]
-    path = config[CONF_PATH]
-    api_key = config[CONF_API_KEY]
-    monitored_conditions = config.get(CONF_MONITORED_CONDITIONS, [])
-    users = config.get(CONF_MONITORED_USERS, [])
-    use_ssl = config[CONF_SSL]
-    verify_ssl = config[CONF_VERIFY_SSL]
-
-    session = async_get_clientsession(hass=hass, verify_ssl=verify_ssl)
-
-    api_client = PyTautulli(
-        api_token=api_key,
-        hostname=host,
-        session=session,
-        verify_ssl=verify_ssl,
-        port=port,
-        ssl=use_ssl,
-        base_api_path=path,
+    """Set up the Modem Caller ID component."""
+    if CONF_MONITORED_CONDITIONS in config:
+        config.pop(CONF_MONITORED_CONDITIONS)
+    if CONF_MONITORED_USERS in config:
+        config.pop(CONF_MONITORED_USERS)
+    if CONF_NAME in config:
+        config.pop(CONF_NAME)
+    if CONF_PORT in config:
+        port = config[CONF_PORT]
+        config.pop(CONF_PORT)
+    if CONF_PATH in config:
+        path = config[CONF_PATH]
+        config.pop(CONF_PATH)
+    protocol = f"http{'s' if config.get(CONF_SSL) else ''}"
+    config[CONF_URL] = f"{protocol}://{config[CONF_HOST]}:{port or '8181'}{path or ''}"
+    config.pop(CONF_HOST)
+    config.pop(CONF_PLATFORM)
```

Comment:
Move this logic to the import step in the config_flow file and use `PyTautulliHostConfiguration` instead of all this popping and construction.